src/net_processing.cpp

```diff
@@ -541,7 +541,7 @@ class PeerManagerImpl final : public PeerManager
     std::map<uint256, std::pair<NodeId, std::list<QueuedBlock>::iterator> > mapBlocksInFlight GUARDED_BY(cs_main);
 
     /** When our tip was last updated. */
-    std::atomic<int64_t> m_last_tip_update{0};
+    std::chrono::seconds m_last_tip_update{0};
```

Comment:
is atomic no longer needed?

Reply:
Initially, I removed it by mistake. But when I tried to initialize var as
```
std::atomic<std::chrono::seconds> m_last_tip_update{0};
```
It is causing a compilation error.

I tried other ways to initialize it, but they don't seem to work.  I searched for a solution on the internet, and I came around this bug report which is in a very similar spirit to this problem: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=54025

So how do you suggest I should proceed now:
- Revert this variable type to `std::atomic<int64_t>`
- Keep the current implementation, which the `std::atomic`

Reply:
`atomic` works for specific concrete integer types, it doesn't work for arbitrary data structures, you'll have to use a full mutex. I don't think this is worth it here so might as well stick with a `int64_t`.

Reply:
chronos are plain integers (they only have a bit of type safety added at compile time).

```cpp
atomic<seconds> a{0s};
```

should compile, like in all other places where `atomic<seconds>` is used.

Reply:
`atomic<seconds> a{0}` was not working but for some reason `atomic<seconds> a{0s}` worked.

Addressed in 1a1fafa4fdcbe7b0d1ef8df466a71bd3e65195d3